src/main/java/org/folio/circulation/domain/validation/ChangeDueDateValidator.java

```diff
@@ -17,33 +20,32 @@ public ChangeDueDateValidator(LoanRepository loanRepository) {
     this.loanRepository = loanRepository;
   }
 
-  public CompletableFuture<Result<LoanAndRelatedRecords>> refuseWhenClaimedReturned(
-      Result<LoanAndRelatedRecords> changedLoanResult) {
+  public CompletableFuture<Result<LoanAndRelatedRecords>> refuseChangeDueDateForItemInStatus(
+    Result<LoanAndRelatedRecords> loanAndRelatedRecordsResult) {
 
-    return changedLoanResult
-      .map(LoanAndRelatedRecords::getLoan)
-      .afterWhen(this::itemIsClaimedReturned,
-        this::refuseWhenDueDateChanged, this::carryOn)
-      .thenApply(r -> r.next(existingLoan -> changedLoanResult));
+    return loanAndRelatedRecordsResult.after(context -> refuseWhenDueDateChanged(context,
+      Loan::isDeclaredLost,
+      Loan::isClaimedReturned,
+      Loan::isAgedToLost));
   }
 
-  private CompletableFuture<Result<Boolean>> itemIsClaimedReturned(
-      Loan changedLoan) {
+  @SafeVarargs
+  private final CompletableFuture<Result<LoanAndRelatedRecords>> refuseWhenDueDateChanged(
+    LoanAndRelatedRecords relatedRecords, Predicate<Loan>... refusePredicates) {
 
-    return Result.ofAsync(() -> changedLoan.getItem().isClaimedReturned());
-  }
+    final Loan changedLoan = relatedRecords.getLoan();
+    final boolean caseDoesNotMatch = Arrays.stream(refusePredicates)
+      .noneMatch(predicate -> predicate.test(changedLoan));
 
-  private CompletableFuture<Result<Loan>> refuseWhenDueDateChanged(
-      Loan changedLoan) {
+    if (caseDoesNotMatch) {
+      return CompletableFuture.completedFuture(succeeded(relatedRecords));
+    }
 
     return getExistingLoan(changedLoan)
       .thenApply(r -> r.failWhen(
         existingLoan -> dueDateHasChanged(existingLoan, changedLoan),
```

Comment:
Why is this request failed when the due date has changed? What have I missed, as I understand it, this operation is specifically to manually change the due date?

I'm asking because I'm finding the name of the calling method to be confusing, as it seems to be trying to do more than one thing.

Reply:
This validator method is used in loan replace endpoint, and the idea is to refuse loan replace process if it is change due date and the loan matches a case (currently item is declared lost or claimed returned or aged to lost)